config/mapprovider.go

```diff
@@ -81,20 +81,39 @@ type ChangeEvent struct {
 
 // Retrieved holds the result of a call to the Retrieve method of a Provider object.
 type Retrieved struct {
+	// Deprecated: Use NewRetrievedFromMap to initialize, and Retrieved.MergeTo to access.
 	*Map
-
-	// CloseFunc specifies a function to be invoked when the configuration for which it was
-	// used to retrieve values is no longer in use and should close and release any watchers
-	// that it may have created.
-	//
-	// If nil, then nothing to be closed.
+	// Deprecated: Use NewRetrievedFromMap to initialize, and Retrieved.Close to access.
 	CloseFunc
 }
 
-// CloseFunc a function to close and release any watchers that it may have created.
+// NewRetrievedFromMap returns a new Retrieved instance that contains a Map data.
+// * cfgMap the Map that will be merged to the given map in the MergeTo.
+// * CloseFunc specifies a function to be invoked when the configuration for which it was
+//   used to retrieve values is no longer in use and should close and release any watchers
+//	 that it may have created.
+func NewRetrievedFromMap(cfgMap *Map, close CloseFunc) Retrieved {
+	return Retrieved{Map: cfgMap, CloseFunc: close}
+}
```

Comment:
@mx-psi last question here, should we go with "close" as "RetrievedOption" because literally is optional here? so we will define a "Option" and we will have `WithClose`, instead of asking users to pass nil when no close needed?

Reply:
Sounds better, yes, it's also how we pass the function pointers on the component helpers, so we already have the precedent there.